lib/invocationHelperFactory.js

```
@@ -0,0 +1,65 @@
+'use strict';
+
+var _ = require('lodash');
+var AWS = require('aws-sdk');
+var logMonitorFactory = require('./logMonitorFactory');
+var logCollectorFactory = require('./logCollectorFactory');
+
+module.exports = function() {
+  var lambda = new AWS.Lambda();
+
+  var invocationHelper = { };
+  var invocationTask = { };
+
+  invocationHelper._onInvokeComplete = function(
+    response,
+    invocationTask,
+    callback
+  ) {
+    invocationTask.requestId =
+      response.httpResponse.headers['x-amzn-requestid'];
+    invocationTask.logMonitor = logMonitorFactory(invocationTask);
+    invocationTask.logCollector = logCollectorFactory(invocationTask);
+
+    callback(null, invocationTask);
+  };
+
+  invocationHelper.invokeLambdaFunction = function(
+    functionName,
+    event,
+    callback
+  ) {
+    invocationTask = {
+      functionName: functionName,
+      event: event,
+      eventList: [ ],
+      logMonitor: undefined,
+      logCollector: undefined,
+      requestId: undefined,
+      startTime: (new Date()).getTime()
+    };
+    var request;
+
+    console.log('Invoking: ' + functionName);
+
+    request = lambda.invokeAsync(
+      {
+        FunctionName: functionName,
+        InvokeArgs: JSON.stringify(event)
+      }
+    );
```

Couldn't you supply `_partialRight(invocationHelper._onInvokeComplete, invocationTask, callback)` as a callback argument to `lambda.invokeAsync` rather than doing the `on('complete')` event handling below. I think if you did that you could also propagate any errors lambda encountered during invocation.